src/operator/elementwise_binary_broadcast_op-inl.h

```diff
@@ -0,0 +1,494 @@
+          Tensor<xpu, 3, DType> mrhs_grad =
+              rhs_grad->get_with_shape<xpu, 3, DType>(ret_reshaped.get<3>(), s);
+          Tensor<xpu, 3, DType> mrhs_data =
+              rhs.data.get_with_shape<xpu, 3, DType>(ret_reshaped.get<3>(), s);
+
+          ASSIGN_DISPATCH(
+              mlhs_grad, req_lhs_grad, sumall_except_dim<1>(mout_grad * mrhs_data));
+          if (lhs_size == 1) {
+            ASSIGN_DISPATCH(mrhs_grad, req_rhs_grad,
+                            mout_grad * broadcast_scalar(mlhs_data, ret_reshaped.get<3>()));
+          } else {
+            ASSIGN_DISPATCH(mrhs_grad, req_rhs_grad,
+                            mout_grad * broadcast<1>(mlhs_data, ret_reshaped.get<3>()));
+          }
+        }
+      } else {
+        if (rhs_broadcast_axis >= 0) {
+          // only broadcast rhs
+          Tensor<xpu, 3, DType> mlhs_grad =
+              lhs_grad->get_with_shape<xpu, 3, DType>(ret_reshaped.get<3>(), s);
+          Tensor<xpu, 3, DType> mlhs_data =
+              lhs.data.get_with_shape<xpu, 3, DType>(ret_reshaped.get<3>(), s);
+          Tensor<xpu, 1, DType> mrhs_grad =
+              rhs_grad->get_with_shape<xpu, 1, DType>(Shape1(rhs_size), s);
+          Tensor<xpu, 1, DType> mrhs_data =
+              rhs.data.get_with_shape<xpu, 1, DType>(Shape1(rhs_size), s);
+          Tensor<xpu, 3, DType> mout_grad =
+              out_grad.data.get_with_shape<xpu, 3, DType>(ret_reshaped.get<3>(), s);
+
+          if (rhs_size == 1) {
+            ASSIGN_DISPATCH(mlhs_grad, req_lhs_grad,
+                            mout_grad * broadcast_scalar(mrhs_data, ret_reshaped.get<3>()));
+          } else {
+            ASSIGN_DISPATCH(mlhs_grad, req_lhs_grad,
+                            mout_grad * broadcast<1>(mrhs_data, ret_reshaped.get<3>()));
+          }
+          ASSIGN_DISPATCH(
+              mrhs_grad, req_rhs_grad, sumall_except_dim<1>(mout_grad * mlhs_data));
+        } else {
+          LOG(FATAL) << "no broadcast is needed";
+        }
+      }
+    });
+}
+
+
+MXNET_REGISTER_SIMPLE_OP(_broadcast_plus, XPU)
+.set_symbol_op_name("BroadcastPlus")
+.set_shape_function(BinaryBroadcastShape_)
+.set_function(XPU::kDevMask, BinaryBroadcastForward_<XPU, mshadow::op::plus>, kInplaceLhsOut)
+.set_gradient(XPU::kDevMask, BinaryBroadcastBackward_<
+              XPU, mshadow_op::identity, mshadow_op::identity>, kInplaceOutLhs)
+.describe("lhs add rhs with broadcast");
+
+MXNET_REGISTER_SIMPLE_OP(_broadcast_minus, XPU)
+.set_symbol_op_name("BroadcastMinus")
+.set_shape_function(BinaryBroadcastShape_)
+.set_function(XPU::kDevMask, BinaryBroadcastForward_<XPU, mshadow::op::minus>, kInplaceLhsOut)
+.set_gradient(XPU::kDevMask, BinaryBroadcastBackward_<
+              XPU, mshadow_op::identity, mshadow_op::negation>, kInplaceOutLhs)
+.describe("lhs minus rhs with broadcast");
+
+MXNET_REGISTER_SIMPLE_OP(_broadcast_mul, XPU)
+.set_symbol_op_name("BroadcastMul")
```

Comment:
for new operators, let us keep the name consistent, with lower cases, so mx.sym and mx.nd have exact the same function.